File: February 2021/13.02.2021/Kaan.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {
        std::vector<TreeNode*> s;
        s.push_back(root);
        int currentLevelCount = 1;
        int answer =root->val;
        while(s.size()!=0){ 
            if(currentLevelCount==0){
                currentLevelCount = s.size();
                answer = s[0]->val;
            }
            if(s[0]->left != nullptr){
                s.push_back(s[0]->left);
            }
            if(s[0]->right != nullptr){
                s.push_back(s[0]->right);
            }
            s.erase(s.begin());
            currentLevelCount--;   
        }      
        return answer;
    }
};

File: February 2021/12.02.2021/Kaan.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        
        List<Integer> list = new ArrayList<>();
        inOrderTraversal(root,list);
        
        return list;
        
        
        
        
    }
    
    public void inOrderTraversal(TreeNode root,List<Integer> list){
        
        if(root == null){
            return;
        }
        else{
            inOrderTraversal(root.left,list);
            list.add(root.val);
            inOrderTraversal(root.right,list);
        }
        
        
    }
}
